### DISCUSSION TITLE
How can vscode be configured to support C++ 's modules?

### DISCUSSION THREAD

**Discussion by masx200** [microsoft/vscode-cpptools#10891] [Q&A]
![1682523398114.jpeg.png](https://user-images.githubusercontent.com/34191203/234627780-49380e90-d621-420a-92ad-c46b1ed5aa00.png)

**Comment by bobbrow** [comment]
C++ modules are not officially supported yet, but it is possible to make IntelliSense work with modules if you are using the MSVC compiler and follow the discussion here: #6302

It will require you to keep your build up to date because we will not automatically build the modules for you.

**Reply by masx200** [reply]
At present, the format of the module IFC has not been standardized, why not let me manually specify the source files or set which directories to search for the source files? And other compilers other than mavc do not support IFC, do other compilers generate module binaries in different formats?

**Reply by bobbrow** [reply]
Questions like this are the reason we don't have official support for modules yet.  MSVC finished their implementation for Visual Studio and worked with the IntelliSense team to get support for the IFC format.  Since we share the IntelliSense implementation with Visual Studio that's why it can currently be made to work with modules for MSVC if you set the right flags and keep your build up to date.
